ts/IdentificationModal.tsx

```diff
@@ -140,8 +160,19 @@ class IdentificationModal extends React.PureComponent<Props, State> {
       );
     } else {
       // if the biometric is not available unlock the unlock code insertion
-      this.setState({ canInsertPin: true });
+      this.setState({ canInsertPinBiometry: true });
     }
+
+    // tslint:disable-next-line: no-object-mutation
+    this.idCheckCanInsertPin = setInterval(this.checkCanInsertPin, 100);
+    this.checkCanInsertPin();
+  }
+
+  public componentWillUnmount() {
+    if (this.idCheckCanInsertPin !== undefined) {
+      clearTimeout(this.idCheckCanInsertPin);
```

Comment:
If I understood right the logic behind this, if an identification fails the app has to:

- check the fails status
- check if the user is in the "free" interval
- if it is not in the "free" interval schedule an update of the time remaining since the next attempt
- stop the previous check when it is no needed

If these points are right, here a mine attempt 93320a147c48458aefcf03bb3539900062a78ce1

Reply:
I've tried your attempt but there is a problem with the following workflow:
Insert wrong pin 5 times -> display lock screen -> close application -> reopen application -> wait for end timer -> insert right pin -> interval method is executed forever.

Starting from your solution, I've slightly changed the approach, using the `remainingAttempts` and `canInsertPinTooManyAttempts` as condition to cancel or start the interval.